optuna/study.py

```diff
@@ -371,6 +371,8 @@ def optimize(
                         )
                         for _ in _iter
                     )
+        except KeyboardInterrupt:
+            self._storage.clean_running_trials()
```

Comment:
Thanks for your suggestion and I agree with you. I've considered this also. The reason I didn't put the handler into the single worker is that, though the `KeyboardInterrupt ` triggered stops all running workers, other workers will start if there're remaining trials to run,  which may cause further changes of other trial states. But this can be solved by just holding down Ctrl-C. So I'll try to move the handler into the worker part.